lib/matplotlib/dviread.py

```diff
@@ -595,11 +595,13 @@ def _height_depth_of(self, char):
                 result.append(0)
             else:
                 result.append(_mul2012(value, self._scale))
-        # cmsy10 glyph 0 ("minus") has a nonzero descent so that TeX aligns
-        # equations properly (https://tex.stackexchange.com/questions/526103/),
-        # but we actually care about the rasterization depth to align the
-        # dvipng-generated images.
-        if self.texname == b"cmsy10" and char == 0:
+        # cmsyXX (symbols font) glyph 0 ("minus") has a nonzero descent
+        # so that TeX aligns equations properly
+        # (https://tex.stackexchange.com/questions/526103/),
+        # but we actually care about the rasterization depth to align
+        # the dvipng-generated images.
+        minus_re = re.compile(br'^cmsy\d+$')
+        if minus_re.match(self.texname) and char == 0:
```

Comment:
Not much point doing these separately if it's not going to be cached.
```suggestion
        if re.match(br'^cmsy\d+$', self.texname) and char == 0:
```

Reply:
:+1:

`re.compile()` and thus also all `re` functions cache generated expressions anyway (AFAIR the cache size is 100). So unless one is excessivlely using regular expressions there is usually no need to explicitly cache compiled expressions.

The only reason one could split this is to give the expression a name, but given the comment above it, that's not necessary either.

Reply:
Thanks for the pointer - seems sensible